app/controllers/api/v1/students_controller.rb

```diff
@@ -3,13 +3,31 @@ class API::V1::StudentsController < API::APIController
   # POST api/v1/students
   def create
     registration = API::V1::StudentRegistration.new(params)
-    if !current_visitor.anonymous?
-      registration.set_user current_visitor
+
+    # This was added to allow for registering after logging in the first time with SSO
+    # But it also occurs if a user is able to access the registration form while being
+    # logged in a different window.
+    if current_user
+      # If the user has a portal_teacher or portal_student, we don't want them re-registering
+      # The errors in this case will be passed down to the registration form.
+      # The use of class_word is so the error message is shown in the form.
+      if current_user.portal_teacher
+        error(class_word: I18n.t('Registration.LoggedInAsTeacher'))
```

Comment:
Yeah I tried no key as well. I was also hoping to see a simple way so this logic wasn't repeated in both the students and teachers controller. But I didn't come up with anything that seemed easy to understand.